core/commands/files/files.go

```diff
@@ -428,10 +435,328 @@ Examples:
 			return buf, nil
 		},
 	},
-	Type: FilesLsOutput{},
+	Type: filesLsOutput{},
+}
+
+type treeItem struct {
+	Name      string
+	Depth     int
+	LastChild bool
+}
+
+type treeSummary struct {
+	Local     bool
+	SizeLocal uint64 `json:",omitempty"`
+	SizeTotal uint64 `json:",omitempty"`
+	NbFile    int    `json:",omitempty"`
+	NbDir     int    `json:",omitempty"`
+}
+
+const itemType = "item"
+const summaryType = "summary"
+
+type treeOutput struct {
+	Hash string
+	Type string
+
+	Item    treeItem
+	Summary treeSummary
+}
+
+var filesTreeCmd = &cmds.Command{
+	Helptext: cmdkit.HelpText{
+		Tagline: "Show informations about a tree of files",
+		ShortDescription: `
+'ipfs files tree' display the tree structure of directories/files
+
+The path can be inside of MFS or not.
+
+Examples:
+
+	$ ipfs files tree /ipfs/QmQLXHs7K98JNQdWrBB2cQLJahPhmupbDjRuH1b9ibmwVa
+`,
+		LongDescription: `
+'ipfs files tree' display the tree structure of directories/files
+
+The path can be inside of MFS or not.
+
+Examples:
+
+	$ ipfs files tree /ipfs/QmQLXHs7K98JNQdWrBB2cQLJahPhmupbDjRuH1b9ibmwVa/locale
+	QmPA5R3e7FJZpFAT5NYRcYuLJay1qS3enu4zUHAkQMu5uW
+	├── webui-cs.json
+	├── webui-de.json
+	├── webui-en.json
+	├── webui-fr.json
+	├── webui-nl.json
+	├── webui-pl.json
+	├── webui-th.json
+	└── webui-zh.json
+	0 directories, 8 files present
+	23 kB present of 23 kB (100.00%)
+`,
+	},
+	Arguments: []cmdkit.Argument{
+		cmdkit.StringArg("path", true, false, "Path to print information about."),
+	},
+	Options: []cmdkit.Option{
+		cmdkit.BoolOption("only-summary", "s", "Only print the summary of the tree's information."),
```

Comment:
That's not a tree. IMO, this doesn't belong in this command.

Reply:
Than maybe this command is misnamed (https://github.com/ipfs/go-ipfs/pull/4596#issuecomment-360018975).  See other comment.